pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md

```diff
@@ -1,21 +1,120 @@
 ```k
-requires "wasm-text.md"
+requires "wasm.md"
 requires "ulm.k"
 ```
 
 ```k
 module ULM-WASM-SYNTAX
-  imports WASM-SYNTAX
+    imports WASM
+```
+
+Program Encoding
+----------------
+
+The ULM-integrated WASM VM has two possible program encodings:
+
+1.  In the local test VM case, a direct encoding is used.
+
+    ```local
+    syntax PgmEncoding ::= ModuleDecl
+    ```
+
+2.  In the remote ULM-integrated VM case, a byte encoding is used.
+
+    ```remote
+    syntax PgmEncoding ::= Bytes
+    ```
+
+```k
 endmodule
 ```
 
 ```k
 module ULM-WASM
-  imports WASM-TEXT
-  imports ULM
+    imports ULM-WASM-SYNTAX
+    imports ULM
+```
 
-  configuration <ulm-wasm>
-                  <wasm/>
-                </ulm-wasm>
+Program Decoding
+----------------
+
+The WASM VM must decode the input program:
+
+1.  In local test VM case, the decoding function is just identity.
+
+    ```local
+    syntax ModuleDecl ::= decodePgm(ModuleDecl) [function, total]
+    // ----------------------------------------------------------
+    rule decodePgm(Mod) => Mod
+    ```
+
+2.  In the remote ULM-integrated VM case, a specialized, hooked byte decoder is used.
+
+    ```remote
+    syntax ModuleDecl ::= decodePgm(Bytes) [function, hook(ULM.decode)]
+    ```
+
+Configuration
+-------------
+
+Here we define the initial configuration.
+Note that the default status code indicates an internal error; this is used defensively, since if we ever get stuck, an error will always be indicated.
+Similarly, we define a default null output which may indicate internal errors.
+
+```k
+    syntax OutputData ::= "NO_OUTPUT"
+                        | Bytes
+
+    configuration
+      <ulmWasm>
+        <k> $PGM:PgmEncoding </k>
+        <wasm/>
+        <create> $CREATE:Bool </create>
+        <entry> $WASM:String </entry>
```

Comment:
How do you plan to use this entry cell?

Also, In the "remote" case, my guess is that the `$WASM` variable will not be set. If that's the case, I'm not sure what it means - will it have a default string value, or will execution fail?